### PR TITLE
Sync code with promql codemirror

### PR DESCRIPTION
**Pull request by aocenas**
Mainly copy pasting code, with some changes so the current structure still works.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,14 +13,14 @@
   "author": "Prometheus Authors <prometheus-developers@googlegroups.com>",
   "license": "Apache-2.0",
   "devDependencies": {
-    "lezer": "^0.13.1",
-    "lezer-generator": "^0.13.1",
+    "@lezer/common": "^0.15.11",
```
Comment (aocenas) [inline package.json:16]: Updated these as they are used in the upstream repo

File: test/test-promql.js [hunk 2]
```diff
@@ -1,17 +1,16 @@
-import {parser} from "../dist/index.es.js"
-import {fileTests} from "lezer-generator/dist/test"
+import { parser } from '../dist/index.es.js';
```
Comment (aocenas) [inline test/test-promql.js:1]: Seems to be mainly small formatting changes

**Review by Nexucis — APPROVED** [review]
nice, thanks !

**Comment by Nexucis** [comment]
@aocenas I just published a new release (0.21.0) so you should be able to upgrade it on your side.

**Comment by aocenas** [comment]
Thank you @Nexucis

**Pull request merged**